library/general/src/lib/config_file/memory_file.rb

```diff
@@ -0,0 +1,17 @@
+module ConfigFile
+  class MemoryFile
+    attr_accessor :content
+
+    def initialize(content = "")
+      @content = content
+    end
+
+    def read(path)
```

Comment:
This being an instance method means you have to pass an instance to a parameter named "class": 
```ruby
memory_file = ConfigFile::MemoryFile.new
config2 = ConfigFile::GrubModel.new(file_class:  memory_file)
```

That's not nice, but I do not see a better option :(